infor-design/enterprise
# 4489 - App Menu Triggers/Role Switcher button accessibility improvements

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR fixes a handful of issues related to accessibility on the Application Menu and Expandable Area and makes them easier to understand by screen readers.

_Application Menu_:
* The Application Menu button now has `aria-controls` and `aria-expanded` information about the app menu.
* Added a localization path for “Application Menu Trigger” which is used in all the examples.

_Expandable Area (Role Switcher feature)_:
* On expandable area, `aria-controls` and `aria-expanded` moved from the header element to the trigger button.  This lets the screen reader actually announce “Show More, Button, Collapsed” and “Show Less, Button, Expanded”.
* On App Menu, the “trigger” setting is passed into the expandable area, defined as the `switchTrigger` button.

_Demoapp_:
* Made basically all app menu trigger buttons point to a Hogan partial instead of being inlined in every sample (a handful weren’t changed for test/demo page consistency).
* Fixed a bug where the icon for the hamburger button wasn’t switching to a close button (X) when the menu opens (switching back was working).
* Fixed the `page-patterns/test-saleshub-external-orders` example.

**Related github/jira issue (required)**:
Closes #4489

**Steps necessary to review your pull request (required)**:
Pull branch, build, run, and open a screen reading tool like Chrome Vox, Apple Voiceover, JAWS, etc.

_Test trigger button improvements:_
- Open http://localhost:4000/components/applicationmenu/example-index
- Turn on the screen reader and focus on the App Menu trigger button (hamburger).  The screen reader should read something like "Application Menu Trigger, selected collapsed, button".
- Click the app menu trigger to open the menu.  Focus the app menu trigger again.  It should say something like "Application Menu Trigger, selected expanded, button".
- Check that after opening the app menu, the hamburger button turned into an "X".

_Test Expandable Area improvements:_
- Open http://localhost:4000/components/expandablearea/example-index.html
- Turn on the screen reader and focus on the first "Show More" link under "Procurement".  The screen reader should read something like "Show More, collapsed, button".
- Click the link to expand.  Focus the link again.  It should say something like "Show More, expanded, button".

_Test Role Switcher:_
- Open http://localhost:4000/components/applicationmenu/example-personalized-role-switcher
- Turn on the screen reader and focus the Role switcher button.  It should say something like "Employee, collapsed, button".
- Click the button to expand the list of roles.  Focus the button again.  It should say something like "Employee, expanded, button".

Also, do a general smoke test of all application menu samples to make sure the App Menu Triggers work, and they have similar accessibility.

Also, check http://localhost:4000/components/page-patterns/test-saleshub-external-vendors.html and make sure it loads without errors.

**Included in this Pull Request**:
